lib/fs.js

```diff
@@ -1380,52 +1380,17 @@ function readdirSyncRecursive(basePath, options) {
   const withFileTypes = Boolean(options.withFileTypes);
   const encoding = options.encoding;
 
-  const readdirResults = [];
-  const pathsQueue = [basePath];
-
-  function read(path) {
-    const readdirResult = binding.readdir(
-      path,
-      encoding,
-      withFileTypes,
-    );
-
-    if (readdirResult === undefined) {
-      return;
-    }
-
-    if (withFileTypes) {
-      // Calling `readdir` with `withFileTypes=true`, the result is an array of arrays.
-      // The first array is the names, and the second array is the types.
-      // They are guaranteed to be the same length; hence, setting `length` to the length
-      // of the first array within the result.
-      const length = readdirResult[0].length;
-      for (let i = 0; i < length; i++) {
-        const dirent = getDirent(path, readdirResult[0][i], readdirResult[1][i]);
-        ArrayPrototypePush(readdirResults, dirent);
-        if (dirent.isDirectory()) {
-          ArrayPrototypePush(pathsQueue, pathModule.join(dirent.parentPath, dirent.name));
-        }
-      }
-    } else {
-      for (let i = 0; i < readdirResult.length; i++) {
-        const resultPath = pathModule.join(path, readdirResult[i]);
-        const relativeResultPath = pathModule.relative(basePath, resultPath);
-        const stat = binding.internalModuleStat(binding, resultPath);
-        ArrayPrototypePush(readdirResults, relativeResultPath);
-        // 1 indicates directory
-        if (stat === 1) {
-          ArrayPrototypePush(pathsQueue, resultPath);
-        }
-      }
-    }
-  }
+  const results = binding.readdirRecursiveSync(
```

Comment:
You need  to also update `typings` folder and add this function to `fs.d.ts` file for intellisense and better IDE experience.